pkg/handler/datastore.go

```diff
@@ -121,6 +122,16 @@ type DataStore interface {
 	GetUpload(ctx context.Context, id string) (upload Upload, err error)
 }
 
+// ServableUpload defines the method for serving content directly
+type ServableUpload interface {
+	ServeContent(ctx context.Context, w http.ResponseWriter, r *http.Request) error
```

Comment:
This function needs proper documentation. That's best done when we have agreed on what the handler does and does not.

Reply:
lmk what/when you want docs on this.